### PR TITLE
Add ray serve's logging context manager

### LINKED ISSUE #15872: [Serve] Lifecycle log should not be INFO
```
(pid=None, ip=172.31.15.157) 2021-05-17 15:22:43,852    INFO on.py:26 -- Waiting for application startup.
(pid=None, ip=172.31.15.157) 2021-05-17 15:22:43,852    INFO on.py:38 -- Application startup complete.
```
Starlette startup events will have these surrounds them, which can be spammy.

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

Current logger is passed into uvicorn's logger which logs with INFO level and a bit spammy. Therefore we can wrap the block for startup / shutdown with slightly higher level within a context manager. 

## Related issue number

Closes #15872 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

========= Use the following local FastAPI wrapper code to repo and verify fixes ==============

```
import requests
from fastapi import FastAPI

import ray
from ray import serve

serve.start()
app = FastAPI()


@serve.deployment(route_prefix="/hello")
@serve.ingress(app)
class MyFastAPIDeployment:
    @app.get("/")
    def root(self):
        return "Hello, world!"

    @app.post("/{subpath}")
    def root(self, subpath: str):
        return f"Hello from {subpath}!"

MyFastAPIDeployment.deploy()
```

=========== Before ==============

```
❯ python logging_fix_stateful.py
2021-06-15 18:16:56,958	INFO services.py:1315 -- View the Ray dashboard at http://127.0.0.1:8265
(pid=87438) 2021-06-15 18:16:58,764	INFO http_state.py:72 -- Starting HTTP proxy with name 'Rhmwjp:SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node:192.168.86.151-0' on node 'node:192.168.86.151-0' listening on '127.0.0.1:8000'
2021-06-15 18:16:58,887	INFO api.py:713 -- Started Serve instance in namespace 'serve'.
2021-06-15 18:16:58,902	INFO api.py:415 -- Updating deployment 'MyFastAPIDeployment'.
(pid=87436) INFO:     Started server process [87436]
(pid=87438) 2021-06-15 18:16:58,975	INFO backend_state.py:773 -- Adding 1 replicas to backend 'MyFastAPIDeployment'.
(pid=87438) 2021-06-15 18:16:58,981	INFO runtime_env.py:107 -- Using preinstalled conda environment: ray
(pid=87465) 2021-06-15 18:17:01,058	INFO on.py:45 -- Waiting for application startup.
(pid=87465) 2021-06-15 18:17:01,058	INFO on.py:59 -- Application startup complete.
```

============ After ==============

```
❯ python logging_fix_stateful.py
2021-06-15 22:10:37,274	INFO services.py:1315 -- View the Ray dashboard at http://127.0.0.1:8265
(pid=96391) 2021-06-15 22:10:39,400	INFO http_state.py:72 -- Starting HTTP proxy with name 'nsXmqR:SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node:192.168.86.151-0' on node 'node:192.168.86.151-0' listening on '127.0.0.1:8000'
2021-06-15 22:10:39,538	INFO api.py:714 -- Started Serve instance in namespace 'serve'.
(pid=96389) INFO:     Started server process [96389]
2021-06-15 22:10:39,553	INFO api.py:416 -- Updating deployment 'MyFastAPIDeployment'.
(pid=96391) 2021-06-15 22:10:39,615	INFO backend_state.py:773 -- Adding 1 replicas to backend 'MyFastAPIDeployment'.
(pid=96391) 2021-06-15 22:10:39,621	INFO runtime_env.py:107 -- Using preinstalled conda environment: ray
```

### CODE REVIEW

**Comment by jiaodong** [comment]
oops latest commit removed json file for some reason -- will walk it through tomorrow, please only check the py files for now.

File: python/ray/serve/utils.py [hunk 1]
```diff
@@ -107,6 +107,74 @@ def parse_request_item(request_item):
...
+        if self.level is not None:
+            self.logger.setLevel(self.old_level)
+        if self.handler:
+            self.logger.removeHandler(self.handler)
+        if self.handler and self.close:
+            self.handler.close()
+        # implicit return of None => don't swallow exceptions
+
+
+class LoggingContext:
```
Comment (simon-mo) [inline python/ray/serve/utils.py:144]: duplicated?

File: python/ray/serve/utils.py [hunk 2]
```diff
@@ -107,6 +107,74 @@ def parse_request_item(request_item):
     return request_item.args, request_item.kwargs
 
 
+class LoggingContext:
+    r"""
```
Comment (simon-mo) [inline python/ray/serve/utils.py:111]: we don't need to r string here, most of the Ray codebase use simple multi-line string for comments.
```suggestion
    """
```
```suggestion
    r"""
```

File: python/ray/serve/utils.py [hunk 3]
```diff
@@ -107,6 +107,74 @@ def parse_request_item(request_item):
...
+    Context manager to manage logging behaviors within a particular block, such as:
+    1) Overriding logging level
+    2) Pass in a logging handler 
+        - Configure if handler should be closed afterwards 
+
+    Source (python3 official documentation)
+    https://docs.python.org/3/howto/logging-cookbook.html#using-a-context-manager-for-selective-logging
+    """
+
+    def __init__(self, logger, level=None, handler=None, close=True):
```
Comment (simon-mo) [inline python/ray/serve/utils.py:121]: let's just drop handler and close? We are not using them and we can always add them back in the future.

File: python/ray/serve/utils.py (hunk 1 above)
Comment (jiaodong) [reply python/ray/serve/utils.py]: yep i kinda messed it up with git amend + manual accept from vscode while trying to resolve conflict, should be good now

**Review by edoakes — APPROVED** [review]

**Review by simon-mo — APPROVED** [review]

**Pull request merged**